Chatto/sources/ChatController/Collaborators/ChatCollectionViewLayout.swift

```diff
@@ -24,8 +24,8 @@
 
 import UIKit
 
-public protocol ChatCollectionViewLayoutDelegate: AnyObject {
-    func chatCollectionViewLayoutModel() -> ChatCollectionViewLayoutModel
+public protocol ChatCollectionViewLayoutModelProviderProtocol: AnyObject {
+    func makeChatCollectionViewLayoutModel() -> ChatCollectionViewLayoutModel
```

Comment:
If it's a 'provider' then it shouldn't 'make' objects. It should 'provide' them. But if it 'makes' them then it should be named a factory or a builder (depends on the pattern you gonna use).